.github/workflows/allthekeeps-ropsten.yml

```diff
@@ -0,0 +1,115 @@
+name: Allthekeeps / Testnet
+
+on:
+  push:
+    branches:
+     - allthekeeps-on-ropsten
+  pull_request:
+  workflow_dispatch:
+    inputs:
+      environment:
+        description: 'Environment for workflow execution'
+        required: false
+        default: 'dev'
+      upstream_builds:
+        description: 'Upstream builds'
+        required: false
+      upstream_ref:
+        description: 'Git reference to checkout (e.g. branch name)'
+        required: false
+        default: 'main'
+
+jobs:
+  dapp-detect-changes:
+    runs-on: ubuntu-latest
+    outputs:
+      path-filter: ${{ steps.filter.outputs.path-filter }}
+    steps:
+      - uses: actions/checkout@v2
+        if: github.event_name == 'pull_request'
+        with:
+          ref: allthekeeps-on-ropsten
+  
+  dapp-build-and-publish:
+    needs: dapp-detect-changes
+    if: |
+      github.event_name != 'pull_request'
+        || needs.dapp-detect-changes.outputs.path-filter == 'true'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: allthekeeps-on-ropsten
+
+      - name: Load environment variables
+        uses: keep-network/ci/actions/load-env-variables@v1
+        with:
+          environment: 'ropsten'
+      
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "12.x"
+
+      - name: Run install repositories
+        run: ./install-repositories.sh
+      
+      - name: Get upstream packages' versions
+        if: github.event_name == 'workflow_dispatch'
+        uses: keep-network/ci/actions/upstream-builds-query@v1
+        id: upstream-builds-query
+        with:
+          upstream-builds: ${{ github.event.inputs.upstream_builds }}
+          query: |
+            keep-core-contracts-version = github.com/keep-network/keep-core/solidity#version
+            keep-ecdsa-contracts-version = github.com/keep-network/keep-ecdsa/solidity#version
+            tbtc-contracts-version = github.com/keep-network/tbtc/solidity#version
+
+      - name: Deploy subgraph to thegraph.com
```

Comment:
Is it safe to deploy the subgraph to `thegraph.com` when workflow gets triggered by `push` or `pull_request`?
If we first run the `allthekeeps-ropsten.yml` as a part of the CI flow (i.e. using `workflow_dispatch` event) we will deploy a subgraph and then install dapp and perform publish to GCP bucket.
And if after some time we crate a PR in the `allthekeeps` project, this will result in deploying the subgraph again (which may differ from the previous one) and we will not push anything new to the GCP registry.
Will that not break the AllTheKeeps? (I'm not fully grasping the role of the subgraph and how is it constructed, so I'm not sure)

Reply:
If we trigger `./deploy-subgraph.sh` it will indeed deploy a new version of subgraph to the `thegraph.com`, however, the version of atk will still point to the previous version (cause the previous version of the 'build' folder was deployed to GCP). `Push` or `pull requests` will validate everything except the deployment to GCP, but it won't break the dApp which was deployed by `workflow_dispatch` event.